File: python/test/misc/test_anagram.py
from unittest import TestCase
from misc.valid_anagram import Anagram

class TestAnagram(TestCase):
    def test_validAnagram(self):
        self.assertTrue(Anagram().isAnagram("anagram", "nagaram"))

    def test_invalidAnagram(self):
        self.assertFalse(Anagram().isAnagram("rat", "cat"))

File: javascript/test/misc/test-anagram.js
'use strict'

var is_anagram = require('../../js/misc/anagram')
var assert = require('assert')

describe('Anagram', function() {
    describe('#is_anagram', function () {
        it('"anagram" and "nagaram" are anagram', function () {
            assert(is_anagram("anagram", "nagaram"), '"anagram" and "nagaram" are supposed to be anagram');
        });

        it('"rat" and "cat" are not anagram', function () {
            assert(!is_anagram("rat", "cat"), '"rat" and "cat" are supposed to be not anagram');
        });
    });
});